### PR TITLE
Add XDG_CONFIG_DIR support

### PR DESCRIPTION
**Pull request by jelly**
Add support for XDG_CONFIG_DIR by adding a dependency to appdir while
keeping compatibility with the existing configuration location.

Closes: #42

### CODE REVIEW

File: lib/urlwatch/filters.py [hunk 1]
```diff
@@ -130,6 +132,9 @@ def __init__(self, job, state):
         super().__init__(job, state)
 
         self.hooks = None
+        if not os.path.exists(self.FILENAME):
+            self.FILENAME = AppDirs('urlwatch').user_config_dir + '/lib/hooks.py'
```
Comment (thp) [inline lib/urlwatch/filters.py]: Use `os.path.join()` instead of hardcoding `/` as path separator.

File: share/man/man1/urlwatch.1 [hunk 2]
```diff
@@ -71,13 +71,13 @@ list supported jobs/filters/reporters
 remove old cache entries
 .SH "FILES"
 .TP
-.B ~/.urlwatch/urls.yaml
+.B ~/.config/urlwatch/urls.yaml
```
Comment (thp) [inline share/man/man1/urlwatch.1]: Well, this should mention `$XDG_CONFIG_DIR`.

File: lib/urlwatch/filters.py (hunk 1 above)
Comment (jelly) [reply lib/urlwatch/filters.py]: Fixed

File: share/man/man1/urlwatch.1 (hunk 2 above)
Comment (jelly) [reply share/man/man1/urlwatch.1]: Okido replaced it with XDG_CONFIG_DIR :)

File: lib/urlwatch/filters.py [hunk 3]
```diff
@@ -130,6 +132,9 @@ def __init__(self, job, state):
         super().__init__(job, state)
 
         self.hooks = None
+        if not os.path.exists(self.FILENAME):
+            self.FILENAME = os.path.join(AppDirs('urlwatch').user_config_dir, 'lib/hooks.py')
```
Comment (thp) [inline lib/urlwatch/filters.py]: I meant:

    self.FILENAME = os.path.join(AppDirs('urlwatch').user_config_dir, 'lib', 'hooks.py')

File: urlwatch [hunk 4]
```diff
@@ -36,9 +36,14 @@ import signal
 import socket
 import sys
 
+from appdirs import AppDirs
+
 pkgname = 'urlwatch'
 urlwatch_dir = os.path.expanduser(os.path.join('~', '.' + pkgname))
 
+if not os.path.exists(urlwatch_dir):
+    urlwatch_dir = AppDirs('urlwatch').user_config_dir
```
Comment (thp) [inline urlwatch]: Use `pkgname` instead of `'urlwatch'`

File: lib/urlwatch/filters.py (hunk 3 above)
Comment (thp) [reply lib/urlwatch/filters.py]: Or actually, wait -- legacy hooks would only be in the older folder, so there's no need to change anything here.

Comment (thp) [reply lib/urlwatch/filters.py]: Meaning if it doesn't exist, don't try to find it in the XDG-based directory.

File: urlwatch (hunk 4 above)
Comment (jelly) [reply urlwatch]: Ah of course, fixed that.

File: lib/urlwatch/filters.py (hunk 3 above)
Comment (jelly) [reply lib/urlwatch/filters.py]: Reverted it all to the original.

**Pull request merged**